.github/workflows/ci.yml

```diff
@@ -95,5 +53,89 @@ jobs:
         # https://buildtracker.dev/docs/guides/github-actions#configuration
         BT_API_AUTH_TOKEN: ${{ secrets.BT_API_AUTH_TOKEN }}
 
+    - name: Upload dist
+      uses: actions/upload-artifact@v1
+      with:
+        name: dist
+        path: dist/
+
     # Fail if any changes were written to source files (ex, from: build/build-cdt-lib.js).
     - run: git diff --exit-code
+
+  # `unit` includes just unit and proto tests.
+  unit:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: git clone
+      uses: actions/checkout@v2
+
+    - name: Use Node.js 10.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 10.x
+
+    - name: Set up protoc
+      uses: arduino/setup-protoc@7ad700d
+      with:
+        version: '3.7.1'
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: 2.7
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install protobuf==3.7.1
+
+    - run: yarn --frozen-lockfile
+
+    - run: yarn test-proto # Run before unit-core because the roundtrip json is needed for proto tests.
+
+    - run: sudo apt-get install xvfb
+    - name: yarn unit
+      run: xvfb-run --auto-servernum yarn unit
+
+  # `smoke` runs as a matrix across 4 jobs:
+  #  * The smoketest groups are split across two runners, to parallelize.
+  #  * Then, those are run with both Chrome stable and ToT Chromium, in parallel
+  smoke:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        chrome-channel: ['stable', 'ToT']
+        smoke-test-invert: [false, true]
+      # e.g. if smoke 0 fails, continue with smoke 1 anyway
+      fail-fast: false
+    env:
+      # The smokehouse tests run by job `smoke_0`. `smoke_1` will run the rest.
+      SMOKE_GROUP_1: a11y oopif pwa pwa2 pwa3 dbw redirects errors offline
```

Comment:
for future reference this can be split up further with one smoke test running `$SMOKE_GROUP_1`, one running `$SMOKE_GROUP_2` and a third running `--invert-match $SMOKE_GROUP_1 $SMOKE_GROUP_2` (and so on with further divisions) but will need to make the matrix dynamically